python/mscclpp/language/tests/test.json

```diff
@@ -0,0 +1,85 @@
+{
+  "name": "allgather",
+  "collective": "allgather",
+  "inplace": true,
+  "protocol": "Simple",
+  "gpus": [
+    {
+      "id": 0,
+      "input_chunks": 1,
+      "output_chunks": 2,
+      "scratch_chunks": 0,
+      "threadblocks": [
+        {
+          "id": 0,
+          "ops": [
+            { "name": "signal", "cids": [0], "ctype": "memory" },
+            { "name": "wait", "cids": [0], "ctype": "memory" },
+            {
+              "name": "put",
+              "src_buff": [ {"type": "o", "index": 0, "size": 1} ],
+              "dst_buff": [ {"buffer_id": 0, "index": 0, "size": 1} ],
+              "cids": [0],
+              "ctype": "memory"
+            },
+            { "name": "signal", "cids": [0], "ctype": "memory" },
+            { "name": "wait", "cids": [0], "ctype": "memory" }
+          ],
+          "channels": [
+            { "channel_type": "memory", "channel_ids": [0] }
+          ],
+          "remote_buffer_refs": [
+            { "access_channel_type": "memory", "remote_buffer_ids": [0] }
+          ]
+        }
+      ],
+      "channels": [
+        { "type": "memory", "connected_to": [1] }
+      ],
+      "remote_buffers": [
+        { "rank": 0, "type": "o", "access_channel_types": ["memory"] }
+      ]
+    },
+    {
+      "id": 1,
+      "input_chunks": 1,
+      "output_chunks": 2,
+      "scratch_chunks": 0,
+      "threadblocks": [
+        {
+          "id": 0,
+          "ops": [
+            { "name": "signal", "cids": [0], "ctype": "memory" },
+            { "name": "wait", "cids": [0], "ctype": "memory" },
+            {
+              "name": "put",
+              "src_buff": [ {"type": "o", "index": 1, "size": 1} ],
+              "dst_buff": [ {"buffer_id": 0, "index": 1, "size": 1} ],
+              "cids": [0],
+              "ctype": "memory"
+            },
+            { "name": "signal", "cids": [0], "ctype": "memory" },
```

Comment:
Do we need to change cids and ctype to channel_ids and channel_types to align with fields in channels